akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -807,13 +810,6 @@ private[akka] class Shard(
   private def receiveTerminated(ref: ActorRef): Unit = {
     if (handOffStopper.contains(ref))
       context.stop(self)
-    else {
-      // workaround for watchWith not working with stash #29101
-      entities.entityId(ref) match {
-        case OptionVal.Some(id) => entityTerminated(id)
-        case _                  =>
-      }
-    }
```

Comment:
perhaps we should log a warning here about receiving an unexpected `Terminated`?

Reply:
Okay. I think i'll also change this a bit to unbreak cinnamon